### DISCUSSION TITLE
可以对server参数同时指定v4和v6的subnet参数吗

### DISCUSSION THREAD

**Discussion by InspoOnU** [pymumu/smartdns#1395] [General]
以下这种写法是否正确？
server-https https://223.5.5.5/dns-query -subnet x.x.x.x/24 -subnet xxxx:xxxx:xxxx:xxxx::/64

**Comment by PikuZheng** [comment]
理论上ipv4是ipv6的子集，所以不行

**Reply by InspoOnU** [reply]
> 理论上ipv4是ipv6的子集，所以不行

了解，二选一填就可以了

**Comment by pymumu** [comment]
目前ecs的设置，ipv6和ipv4是分开的，ipv4的ECS只管A记录，IPv6的ECS只管AAAA记录。
也就是
server-https https://223.5.5.5/dns-query -subnet x.x.x.x/24 -subnet xxxx:xxxx:xxxx:xxxx::/64
这个写法才是目前同时支持IPv4和V6工作

但当前smartdns的ecs工作方式可能有错误。
原理上-subnet的ip地址是让服务端知道客户端是归属那个运营商，那个地区。这里是不区分IPv4或IPv6的。
也就是拿IPV4的ECS，查询AAAA地址，也应该是就近给结果。

smartdns可能要修改改成：
如果单独配置了IPv4，则所有查询的ECS都用IPv4.
如果单独配置了IPv6，则所有查询的ECS都用IPv6.
如果同时配置了IPV4和IPV6，则A记录用IPV4，AAAA记录用IPV6
